webpush.go

```diff
@@ -162,7 +161,7 @@ func SendNotification(message []byte, s *Subscription, options *Options) (*http.
 	ciphertext := gcm.Seal([]byte{}, nonce, plaintext, nil)
 
 	// POST request
-	req, err := http.NewRequest("POST", s.Endpoint, ioutil.NopCloser(bytes.NewReader(ciphertext)))
+	req, err := http.NewRequest("POST", s.Endpoint, bytes.NewReader(ciphertext))
```

Comment:
Why does this matter here? I've been Googling and found https://medium.com/@xoen/golang-read-from-an-io-readwriter-without-loosing-its-content-2c6911805361 but I'm not sure why this sets the ContentLength but the other doesn't?

Reply:
https://github.com/SherClockHolmes/webpush-go/pull/13#discussion_r180233903 might help answer this (particularly the quoted bit)